src/main/java/com/alipay/oceanbase/rpc/ObTableClient.java

```diff
@@ -1337,40 +1337,34 @@ private String refreshTableNameByTableGroup(String physicalTableName, String tab
     /**
      * 根据 rowkey 获取分区 id
      * @param tableEntry
-     * @param rowKey
+     * @param row
      * @return
      */
-    private long getPartition(TableEntry tableEntry, Object[] rowKey) {
-        // non partition
-        if (!tableEntry.isPartitionTable()
-            || tableEntry.getPartitionInfo().getLevel() == ObPartitionLevel.LEVEL_ZERO) {
-            return 0L;
-        }
-
+    private long getPartition(TableEntry tableEntry, Row row) {
         if (tableEntry.getPartitionInfo().getLevel() == ObPartitionLevel.LEVEL_ONE) {
-            return tableEntry.getPartitionInfo().getFirstPartDesc().getPartId(rowKey);
+            return tableEntry.getPartitionInfo().getFirstPartDesc().getPartId(row);
         }
 
-        Long partId1 = tableEntry.getPartitionInfo().getFirstPartDesc().getPartId(rowKey);
-        Long partId2 = tableEntry.getPartitionInfo().getSubPartDesc().getPartId(rowKey);
+        Long partId1 = tableEntry.getPartitionInfo().getFirstPartDesc().getPartId(row);
```

Comment:
There is only one usage for this getPartition function which is in getTable function at line 1455. In the getTable function, it will check whether the table is a partition table. If it's not a partition table, it will directly assign 0L to the partId. So I think no-partition tables will not step into this getPartition function.

Reply:
already changed.